### ISSUE TITLE
Console hooks are broken

### ISSUE THREAD

**Issue by nhocki** [rweng/pry-rails#74]
Hi,

I'm trying to add default methods to the console like this:

```ruby
# config/application.rb
# Source http://opensoul.org/2012/11/08/add-helper-methods-to-your-rails-console/
console do
  require 'console_foo'
  Rails::ConsoleMethods.include(ConsoleFoo)
end
```

Rails [includes][1] `Rails::ConsoleMethods` into the  `console::ExtendCommandBundle` module, in this case that'd be `Pry::ExtendCommandBundle` which is an [empty module](https://github.com/pry/pry/blob/ee2513bdb43ad272bba6895645ce04bb2c3d6377/lib/pry.rb) so nothing is happening.

I could get this working by doing `TOPLEVEL_BINDING.eval('self').extend ConsoleFoo` but that seems a lot like a hack, is there a way we could have `ExtendCommandBundle` work?

The `console` block gets called before `console.start` (`Pry.start`) so maybe there's something to be done there?

Thanks

[1]: https://github.com/rails/rails/blob/7fb45a6130598703b3da8fe9ef29a5c7a1eb9d09/railties/lib/rails/commands/console.rb#L62-L64

**Comment by mlangenberg** [comment]
Just ran into this as well. Can confirm that switching back to IRB mitigates the problem.
When you run `include Rails::ConsoleMethods` in the console, the methods are available again.

```ruby
class Application < Rails::Application
    module ConsoleHelpers
      def bar; :bar; end
    end
    console do
      module Rails::ConsoleMethods
        include ConsoleHelpers
        def foo; :foo; end
      end
    end
end
```

```
rails c
Loading development environment (Rails 4.1.6)
[1] pry(main)> foo
=> :foo
[2] pry(main)> bar
NameError: undefined local variable or method `bar' for main:Object
from (pry):2:in `__pry__'
```

Funny right?

**Comment by nhocki** [comment]
@mlangenberg yes. Going back to plain IRB fixes the problem. I actually ended up extending `TOPLEVEL_BINDING`.  I blogged about this [here].(http://blog.nhocki.com/2015/06/03/add-helper-methods-to-your-rails-console-with-pry/)

**Comment by mlangenberg** [comment]
Thanks for documenting your workaround @nhocki :+1:
